docs/knowledgebase/advanced/try-runtime.md

```diff
@@ -0,0 +1,185 @@
+---
+title: try-runtime
+---
+
+The `try-runtime` tool is built to query a snapshot of runtime storage, using the externalities of a Substrate runtime to store state. In this way, 
+it enables runtime engineers to write tests for a specified runtime state.
+Being part of [`sc-cli`][sc-cli-rustdocs], it is designed to be used as a command line interface to specify 
+what block to capture and which pallets to run tests on.
+
+In its simplest form, `try-runtime` is a tool that enables:
+
+1. Connecting to a remote node and calling into some runtime API.
+2. Taking some or all of the data for that node at a specified block.
+3. Writing tests for that data.
+
+> **Note:** This tool is designed to be used for testing _before_ going to production.
+
+## Motivation
+
+The initial motivation for `try-runtime` came from the need to extend the capabilities of [`TestExternalities`][testextern-rustdocs] and [`BasicExternalities`][basicextern-rustdocs] 
+in such a way that it would be possible to use externalities to populate its memory with real node data. Since externalities provide the capability of storing an in-memory
+hash-map, extending these capabilities is achieved by simply using the key-value pairs from a node's RPCs.
+
+## How it works
+The `try-runtime` tool has its own implementation of externalities called [`remote_externalities`][remoteextern-rustdocs], with its hash-map containing:
+
+- a **Key**: the `Hash` of `hash(pallet_name) + hash(storage)`; and
+- a **Value**: some SCALE encoded `Vec<u8>` Rust value. 
```

Comment:
Some isseus here: 

`RemoteExternalities` is just a builder wrapper around `TestExternalities`.

Keys are not ALWAYS what you said, they sometimes are. How keys are generated is its own topic and you should link to it if it has a page.